pipeline/builder.go

```diff
@@ -37,11 +38,18 @@ func build(c pipelineBuilderOptions) (Pipeline, error) {
 	}
 
 	switch {
-	case c.batchDecoder != nil:
-		setupBatchProcessor(c, p)
-	case c.decoder != nil:
-		setupProcessor(c, p)
+	case c.messagePool != nil:
+		err = setupMessagePool(c, p)
+	case len(c.inputStreams) > 0:
+		err = setupStreamProcessor(c, p)
```

Comment:
Nit: since those seems to be mutually exclusive and now there's a error check, it might be interesting to add a check for "both configured"

Reply:
I see now that there is a similar check on a higher level. This seems to be a sanity check. It could be documented or made explicit in the error message, but this is a nit of a nit thought